### PR TITLE
[ci][mac/2] move all python mac tests to civ2

### PR DESCRIPTION
**Pull request by can-anyscale**
Move all python mac tests to civ2. I also add macos_wheel as a tag so this PR doesn't need to run the whole test suite

Test:
- CI

### CODE REVIEW

**Comment by aslonnie** [comment]
premerge tests seem broken?

File: .buildkite/pipeline.macos.yml [hunk 1]
```diff
@@ -110,20 +110,6 @@ steps:
...
-    - *prelude_commands
-    - TORCH_VERSION=1.9.0 ./ci/env/install-dependencies.sh
-    # Use --dynamic_mode=off until MacOS CI runs on Big Sur or newer. Otherwise there are problems with running tests
-    # with dynamic linking.
-    - bazel test --config=ci --dynamic_mode=off --test_env=CI $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=-post_wheel_build --
-      //:all python/ray/dashboard/... -python/ray/serve/... -rllib/... -core_worker_test
-    - *epilogue_commands
-
-
 - label: ":mac: :apple: Ray C++ and Java"
```
Comment (aslonnie) [inline .buildkite/pipeline.macos.yml:113]: this test is still here? not entirely "the rest of mac tests"? :)

**Review by aslonnie — COMMENTED** [review]
the change looks okay; not sure why tests are failing.

**Comment by can-anyscale** [comment]
There are two failing osx test remaining, one is failing on master (osx://python/ray/tests:test_actor_client_mode) and one is flaky on master and also fail with different test functions on retries (osx://python/ray/tests:test_plasma_unlimited)

File: .buildkite/pipeline.macos.yml (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/pipeline.macos.yml:113]: ah yes not everything yet

**Comment by can-anyscale** [comment]
@rickyyx , @rkooo567 , @stephanie-wang with this migration, osx test jobs now also turn red where there are test failures

**Comment by aslonnie** [comment]
> @rickyyx , @rkooo567 , @stephanie-wang with this migration, osx test jobs now also turn red where there are test failures

can we mark them as soft fail on civ2 postmerge until we have it fixed? (or until we have similar flaky/failing test separation)?

**Review by aslonnie — APPROVED** [review]

**Pull request merged**